hack/generator/pkg/codegen/resource_registration_file.go

```diff
@@ -99,7 +108,7 @@ func (r *ResourceRegistrationFile) AsAst() (*dst.File, error) {
 func (r *ResourceRegistrationFile) generateImports() (*astmodel.PackageImportSet, error) {
 	requiredImports := astmodel.NewPackageImportSet()
 
-	for _, typeName := range r.resources {
+	for _, typeName := range append(r.resources, r.storageVersionResources...) {
```

Comment:
Perhaps use `PackageImportSet.versionedNameForImport()` to create the unique names, instead of reinventing the wheel?

Reply:
Promoted that method to public and used it instead. I tried to use the `ResolveConflicts` method but that didn't get the same formatting as none of the names conflicted -- since we know the form of all of the imports in this file it made sense to me to force them to be named "nicely" (which is what the `VersionedNameForImport` does).

Take a look and see if you like it.

Reply:
I moved the `VersionedNameForImport` and `ServiceNameForImport` to `PackageImport` rather than the set to make this easier for me to do (I think they make more sense there anyway as it's an operation on the import not on a set)

Reply:
Looks good. :grin: